keanu-project/src/main/java/io/improbable/keanu/algorithms/mcmc/NUTS.java

```diff
@@ -77,6 +78,7 @@ public NetworkSamples getPosteriorSamples(final BayesianNetwork bayesNet,
                                               final List<? extends Vertex> sampleFromVertices,
                                               final int sampleCount) {
 
+        Preconditions.checkArgument(!sampleFromVertices.isEmpty(), "List of vertices to sample from is empty");
```

Comment:
Just curious, why was this added now?

Reply:
We were doing some mocking for testing and the realised that NUTS doesn't terminate if you pass in an empty list of vertices. So added the check in case someone else runs into this issue.